### PR TITLE
Initial vLLM-QAIC plugin for Qualcomm Cloud AI Accelerators

### PR DESCRIPTION
**Pull request by ankitqti**
Introduces the vLLM-QAIC plugin with V1 architecture support, enabling both Ahead-of-Time (AOT) compiled and PyTorch JIT execution modes.

AOT Execution Mode:
- Generic model execution (LLM, MoE, Embedding, Reranker, Score)
- QPC execution flow and QEFF model compilation
- CCL support for pipeline prefill
- Speculative decoding (Ngram, Suffix, Draft)
- LoRaX adapter serving (Dynamic & Fixed)
- Disaggregated Serving:
  - Pipeline prefill execution with Tensor Splitting
  - MDP Partitioner support
  - QAIC shared memory connector
  - LMCache connector
  - Disaggregated orchestration package integration
  - Grafana/Prometheus monitoring dashboard
  - LLM-D support
  - Auto head replication
  - KV cache transfer for heterogeneous attention layers
- Vision-Language Models:
  - 2QPC and 3QPC VLM execution
  - Qwen3VL, Qwen3VLMoE, InternVL, Qwen2.5VL
- Scheduling & Performance:
  - Scheduler optimization with async scheduling
  - Chunked prefill scheduling (reduces P99 TPOT)

PyTorch (JIT) Execution Mode:
- PyTorch eager mode execution
- Fused Kernels (RMSNorm, TopK Softmax, Grouped TopK, TopK Sigmoid)

This plugin represents the culmination of two years of development by the Qualcomm Datacenter LLM Serving team, transitioning from an internally maintained vLLM fork to a modular plugin-based architecture.

Contributors

Ankit Arora <ankaro@qti.qualcomm.com>
Sanidhya Singal <sanising@qti.qualcomm.com>
Xiyue Shi <xiyushi@qti.qualcomm.com>
Shivani Athavale <athavale@qti.qualcomm.com>
Erick Platero <eplatero@qti.qualcomm.com>
Mahesh Balasubramanian <mahebala@qti.qualcomm.com>
Animesh Kumar <animku@qti.qualcomm.com>
Apoorva Gokhale <agokhale@qti.qualcomm.com>
Gokkulnath Thirukonda Srinivasan <gthiruko@qti.qualcomm.com>
Jou-An Chen <jouachen@qti.qualcomm.com>
Varun Gupta <vargupt@qti.qualcomm.com>
Abdul Rasheed Sahni <sahni@qti.qualcomm.com>
Mohit Mehta <mohmeh@qti.qualcomm.com>
Neeharika Gupta <neeharik@qti.qualcomm.com>
Mrunal Kshirsagar <mkshirsa@qti.qualcomm.com>
Aditya Dhananjay Jadhav <aditjadh@qti.qualcomm.com>
Mohit Gupta <guptam@qti.qualcomm.com>
Sahil Deshmukh <sahidesh@qti.qualcomm.com>
Vahid Janfaza <vjanfaza@qti.qualcomm.com>
Chaitanya Sachdeva <csachdev@qti.qualcomm.com>
Shriniwas Kulkarni <shriniwa@qti.qualcomm.com>

### CODE REVIEW

**Comment by ankitqti** [comment]
All looks good !
